### PR TITLE
trans: Use an isize to count the number of registers so we don't underflow for fn's with > 7 args in debug builds.

### LINKED ISSUE #29073: Arithmetic operation overflowed in rustc_trans::trans::cabi_x86_64
I'm getting an "arithmetic operation overflowed" ICE in librustc_trans/trans/cabi_x86_64.rs on line 454 when building stage1 libstd on 64-bit Ubuntu. Likely related to #27017.

### PR DESCRIPTION
**Pull request by luqmana**
Fixes #29073.

### CODE REVIEW

**Comment by pnkfelix** [comment]
@luqmana btw, were you testing/developing this in a debug build?  (Probably not since bootstrapping debug-builds is [currently hosed](https://github.com/rust-lang/rust/issues/28947).)

I ask because this seems like a case we could add to the list of "bugs that would be caught by overflow/underflow checking arithmetic"...

File: src/test/run-make/extern-fn-struct-passing-abi/test.c [hunk 1]
```diff
@@ -59,6 +59,27 @@ void byval_rect(int32_t a, int32_t b, int32_t c, int32_t d, int32_t e, struct Re
...
 
 // System V x86_64 ABI:
+// a, b, c, d, e, f should be in registers
+// s should be byval pointer on the stack
+//
+// Win64 ABI:
+// a, b, c, d should be in registers
+// e, f should be on the stack
+// s should be byval pointer on the stack
+void byval_many_rect(int32_t a, int32_t b, int32_t c, int32_t d, int32_t e, int32_t f, struct Rect s) {
```
Comment (apasel422) [inline src/test/run-make/extern-fn-struct-passing-abi/test.c]: This line failed tidy (> 100 chars).

**Comment by luqmana** [comment]
@pnkfelix Yea, it was with just `--enable-optimize` i believe since op+debug was broken and just debug too long.

**Comment by apasel422** [comment]
Looks like this timed out.

**Comment by dotdash** [comment]
@bors retry
Am 17.10.2015 14:35 schrieb "Andrew Paseltiner" <notifications@github.com>:

**Comment by apasel422** [comment]
It doesn't seem like the test failure is related to this PR. I was able to compile and run all tests successfully on my local 64-bit Linux machine with optimizations enabled.

**Pull request merged**